internal/provider/newrelic/provider_integration_test.go

```diff
@@ -111,8 +115,6 @@ func Test_Getting_external_metric_generates_a_query_not_rejected_by_backend(t *t
 			selector := selector()
 
 			t.Run(testCaseName, func(t *testing.T) {
-				t.Parallel()
```

Comment:
I needed to remove it because there were conflicts while using the proxy; my guess is because of environment variables 😞

Reply:
ohug! yes it should be that , what about using MainTest to orchestrate them WDYT?

Reply:
Shouldn't be fixed with https://github.com/golang/go/issues/41260? 
-> https://go-review.googlesource.com/c/go/+/260577/

I see that you are using it already, therefore I am a bit lost here 😕

Reply:
ahh no, it says specifically:
```
// Setenv calls os.Setenv(key, value) and uses Cleanup to
// restore the environment variable to its original value
// after the test.
//
// This cannot be used in parallel tests.
```

Reply:
But still, this test should be not affected by the other ones if they are not run in parallel

Reply:
> But still, this test should be not affected by the other ones if they are not run in parallel

Yes, this is what the logic says, parallel tests are run in parallel only with other parallel tests... But maybe these parallel tests are run in another goroutine, while let's say the main goroutine is running the proxy tests sequentially. So, setting the env variable is global and when the parallel test tries to get the external metrics is trying to use the proxy because of the env var global scope, ending up with a connection refused.

Reply:
> ohug! yes it should be that , what about using MainTest to orchestrate them WDYT?

AFAIK `MainTest` let you orchestrate the tests by adding some generic before and after behaviours in a per package way, but doesn't allow you to access each individual test per se; so the problem would be the same I think.